benchmark/distributed_join.cu

```diff
@@ -158,7 +158,7 @@ int main(int argc, char *argv[])
     Communicator* communicator;
     if (COMMUNICATOR_NAME == "UCX") {
         communicator = initialize_ucx_communicator(
-            USE_BUFFER_COMMUNICATOR, 2 * mpi_size, 800'000'000LL / mpi_size - 100'000LL
+            USE_BUFFER_COMMUNICATOR, 2 * mpi_size * 4, 200'000'000LL / mpi_size - 100'000LL
```

Comment:
Does this depend on the overdecomposition factor and the problem size? We should try to use less "magic" numbers here in the formula. And if we do use some literals, we should document it better.

Reply:
It does not. 4 is because we have 4 columns across both the build and probe table, and 2 is because we need to both send and receive the buffer. I will document this.

Reply:
*4 is no longer needed after 4f7f099.